### PR TITLE
Use register for default types

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Use a mechanism to register the default types (only a POC - can not be merged).
Fix some warnings regarding tests

## Affected Dependencies
Register primitives using a factory method (this should remove circular dependencies)

## How has this been tested?
- Tests should be work

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [X] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
Is this to help automatically boxing python primitives into SyPrimitive?

**Review by iamtrask — APPROVED** [review]
I like the style in which this is done

File: src/syft/lib/python/primitive_factory.py [hunk 1]
```diff
@@ -24,44 +26,74 @@ def isprimitive(value: Any) -> bool:
...
+    @syft_decorator(typechecking=True)
+    def register_primitive(
+        python_primitive: Union[
+            Type[int],
+            Type[float],
+            Type[bool],
+            Type[complex],
+            Type[list],
+            Type[str],
+            None,
```
Comment (madhavajay) [inline src/syft/lib/python/primitive_factory.py:85]: Looks like we can use Type[None] to get typing.Type[NoneType]

Comment (madhavajay) [reply src/syft/lib/python/primitive_factory.py:85]: Actually, I remember trying to resolve this and just got into a loop of mypy errors that make no sense, so feel free to leave it as is.

**Comment by madhavajay** [comment]
@gmuraru I think maybe we can do this type of thing as discussed on Slack:
https://github.com/OpenMined/PySyft/pull/4637

Then maybe we can just close this for now and re-discuss in future if theres a desire to go down this path again?

**Pull request closed without merging**